### PR TITLE
Pruning / Updating Doxygen 

### PR DESCRIPTION
**Pull request by felipeblazing**
The only thing this originally wanted to do is add a comment to the signature of gdf_augmented_view. 

In this process we noticed that there were doxygen comments on both function declarations and function implementations. Removed comments from implementations and updated declarations when appropriate. Also added some more doxygen comments to a few structs and enums in cpp/include/cudf/types.h

This depends on https://github.com/rapidsai/cudf/pull/823

### CODE REVIEW

File: cpp/include/cudf/functions.h [hunk 1]
```diff
@@ -90,6 +90,7 @@ gdf_error gdf_column_view(gdf_column *column, void *data, gdf_valid_type *valid,
  * @param[in] size Number of rows in the column.
  * @param[in] dtype Data type of the column.
  * @param[in] null_count The number of non-valid elements in the validity bitmask
+ * @param[in] extra_info Extra information for column description
```
Comment (jrhemstad) [inline cpp/include/cudf/functions.h]: I'd add a reference to see the definition/documentation of the structure for more detail.

Comment (jrhemstad) [reply cpp/include/cudf/functions.h]: (And make sure the documentation for the struct is up-to-date.)

File: cpp/include/cudf/types.h [hunk 2]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ * enum allows us to define different units of time. The types included are nanosecond,
+ * microsecond, millisecond, and second.
+ */
 typedef enum {
-	TIME_UNIT_NONE=0, // default (undefined)
-	TIME_UNIT_s,   // second
-	TIME_UNIT_ms,  // millisecond
-	TIME_UNIT_us,  // microsecond
-	TIME_UNIT_ns   // nanosecond
+	TIME_UNIT_NONE=0, /**< The default time unit type. */
```
Comment (harrism) [inline cpp/include/cudf/types.h]: Please use `///<` for single line docs rather than C-style comments.

File: cpp/include/cudf/types.h [hunk 3]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+	TIME_UNIT_s,   /**< Second resolution time unit type */
+	TIME_UNIT_ms,  /**< Millisecond resolution time unit type */
+	TIME_UNIT_us,  /**< Microsecond resolution time unit type */
+	TIME_UNIT_ns   /**< Nanosecond resolution time unit type */
 } gdf_time_unit;
 
+/**
+ * @brief This struct allows us to include extra information about a columns type.
```
Comment (harrism) [inline cpp/include/cudf/types.h]: Please avoid the "royal we" (and us). "Extra information about column type" is sufficient here.

File: cpp/include/cudf/types.h [hunk 4]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ *
+ * Column types sometimes need extra information to properly encode the representation
```
Comment (harrism) [inline cpp/include/cudf/types.h]: These three lines add no useful information, are verbose, and use the "royal us". Delete them and document the meaning of `time_unit`, preferably with an inline `///<`.

File: cpp/include/cudf/types.h [hunk 5]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ * us to know the resolution of a time type.
+ */
 typedef struct {
 	gdf_time_unit time_unit;
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+
+/**
+ * @brief Defines the C representation of a column in CUDF. This is the main unit of operation.
```
Comment (harrism) [inline cpp/include/cudf/types.h]:
```suggestion
 * @brief The C representation of a column in CUDF. This is the main unit of operation.
```

File: cpp/include/cudf/types.h [hunk 6]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ *
+ * This struct contains pointers to GPU memory and metadata which describes how that data is layed out.
```
Comment (harrism) [inline cpp/include/cudf/types.h]:
```suggestion
 * This struct contains pointers to GPU memory and metadata to describe data layout.
```

File: cpp/include/cudf/types.h [hunk 7]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ *
+ *
+ */
 typedef struct gdf_column_{
-    void *data;                       /**< Pointer to the columns data */ 
+    void *data;                       /**< Type Erased pointer to the columns data */
```
Comment (harrism) [inline cpp/include/cudf/types.h]:
```suggestion
    void *data;                       /**< Type Erased pointer to the column data */
```

File: cpp/src/dataframe/column.cpp [hunk 8]
```diff
@@ -34,24 +34,7 @@ gdf_error gdf_mask_concat(gdf_valid_type *output_mask,
...
- *             validity bitmasks
- * @param[in] columns_to_concat[] The columns to concatenate
- * @param[in] num_columns The number of columns to concatenate
- * 
- * @return gdf_error GDF_SUCCESS upon completion; GDF_DATASET_EMPTY if any data
- *         pointer is NULL, GDF_COLUMN_SIZE_MISMATCH if the output column size
- *         != the total size of the input columns; GDF_DTYPE_MISMATCH if the
- *         input columns have different datatypes.
- * ---------------------------------------------------------------------------**/
+
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]: Leave at least the brief part of the comment on each function, so readers don't have to jump to the header to know what the function does. Just don't put it in doxygen tags (No `/**` or `@brief`)

File: cpp/doxygen/Doxyfile [hunk 9]
```diff
@@ -0,0 +1,2427 @@
+# Doxyfile 1.8.11
```
Comment (harrism) [inline cpp/doxygen/Doxyfile]: Didn't @jrhemstad 's PR #823 already add this file?

File: cpp/CMakeLists.txt [hunk 10]
```diff
@@ -266,6 +266,16 @@ add_custom_command(OUTPUT INSTALL_PYTHON_CFFI
...
 
+	###################################################################################################
+# - make documentation ----------------------------------------------------------------------------
+
+add_custom_command(OUTPUT CUDF_DOXYGEN
+                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doxygen
+                   COMMAND doxygen Doxyfile
+                   VERBATIM)
+
+add_custom_target(doc DEPENDS CUDF_DOXYGEN)
```
Comment (harrism) [inline cpp/CMakeLists.txt]: I think @jrhemstad added a `make doc` target in another outstanding PR. #823 ?

File: cpp/include/cudf/types.h [hunk 11]
```diff
@@ -72,27 +72,50 @@ typedef enum {
     GDF_HASH_IDENTITY,  /**< Identity hash function that simply returns the key to be hashed */
 } gdf_hash_func;
 
+
+/**
```
Comment (harrism) [inline cpp/include/cudf/types.h]:
```suggestion
/** ---------------------------------------------------------------------------*
```

Comment (harrism) [reply cpp/include/cudf/types.h]: For consistency.

File: cpp/include/cudf/types.h [hunk 12]
```diff
@@ -72,27 +72,50 @@ typedef enum {
...
+ * @brief Defines the unit of time that an algoithm or structure is storing.
+ *
+ * There are time types in cudf. Those time types can have different resolutions so this
+ * enum allows us to define different units of time. The types included are nanosecond,
+ * microsecond, millisecond, and second.
+ */
```
Comment (harrism) [inline cpp/include/cudf/types.h]:
```suggestion
 --------------------------------------------------------------------------- */
```

Comment (harrism) [reply cpp/include/cudf/types.h]: And etc. on all comment blocks, to match existing comment blocks.

Comment (harrism) [reply cpp/include/cudf/types.h]: Now that I look I see we're not consistent with these horizontal lines anymore... So take your pick.

File: cpp/CMakeLists.txt (hunk 10 above)
Comment (felipeblazing) [reply cpp/CMakeLists.txt]: This has his pr merged into it. Is that terrible? I wanted to make sure my changes work I can remove the dependency on his pr if you like.

File: cpp/doxygen/Doxyfile (hunk 9 above)
Comment (felipeblazing) [reply cpp/doxygen/Doxyfile]: yes it has his changes

File: cpp/src/dataframe/column.cpp (hunk 8 above)
Comment (harrism) [reply cpp/src/dataframe/column.cpp]: What I do is take the contents of @brief from the header, and place it above the function in a regular old `//`.  Note doxygen ignores all comments that are not inside a oxygen comment indicator. Those are either `///` or `/** ... */`.

So if you don't have a triple slash or a double asterisk then doxygen ignores it.

File: cpp/include/cudf/functions.h [hunk 13]
```diff
@@ -106,17 +107,24 @@ gdf_error gdf_column_view_augmented(gdf_column *column, void *data, gdf_valid_ty
  */
 gdf_error gdf_column_free(gdf_column *column);
 
-/** 
- * @brief  Concatenates the gdf_columns into a single, contiguous column,
- * including the validity bitmasks
+/** ---------------------------------------------------------------------------*
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: Missed one.

```suggestion
/**
```

File: cpp/include/cudf/functions.h [hunk 14]
```diff
@@ -106,17 +107,24 @@ gdf_error gdf_column_view_augmented(gdf_column *column, void *data, gdf_valid_ty
...
+ * @param[in] columns_to_concat[] The columns to concatenate
  * @param[in] num_columns The number of columns to concatenate
  * 
- * @returns GDF_SUCCESS upon successful completion
- */
+ * @return gdf_error GDF_SUCCESS upon completion; GDF_DATASET_EMPTY if any data
+ *         pointer is NULL, GDF_COLUMN_SIZE_MISMATCH if the output column size
+ *         != the total size of the input columns; GDF_DTYPE_MISMATCH if the
+ *         input columns have different datatypes.
+ * ---------------------------------------------------------------------------**/
```
Comment (harrism) [inline cpp/include/cudf/functions.h]:
```suggestion
 */
```

File: cpp/src/dataframe/column.cpp (hunk 8 above)
Comment (felipeblazing) [reply cpp/src/dataframe/column.cpp]: ok ill add something like this ```/* 
 * @brief Concatenates multiple gdf_columns into a single, contiguous column,
 * including the validity bitmasks.
 */```

Comment (felipeblazing) [reply cpp/src/dataframe/column.cpp]: ugh the code thing isnt working but `/* */ style`

File: cpp/src/dataframe/column.cpp [hunk 15]
```diff
@@ -142,26 +128,12 @@ gdf_error gdf_column_concat(gdf_column *output_column, gdf_column *columns_to_co
   return GDF_SUCCESS;
 }
 
-/** ---------------------------------------------------------------------------*
- * @brief Return the size of the gdf_column data type.
- *
- * @return gdf_size_type Size of the gdf_column data type.
- *  ---------------------------------------------------------------------------**/
+//@brief Return the size of the gdf_column data type.
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
// Return the size of the gdf_column data type.
```

File: cpp/src/dataframe/column.cpp [hunk 16]
```diff
@@ -142,26 +128,12 @@ gdf_error gdf_column_concat(gdf_column *output_column, gdf_column *columns_to_co
...
- *        datatype
- *
- * @param[out] column The output column.
- * @param[in] data Pointer to data.
- * @param[in] valid Pointer to validity bitmask for the data.
- * @param[in] size Number of rows in the column.
- * @param[in] dtype Data type of the column.
- * @return gdf_error Returns GDF_SUCCESS upon successful creation.
- * ---------------------------------------------------------------------------**/
+//@brief  Constructor for the gdf_context struct
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
// Constructor for the gdf_context struct
```

File: cpp/src/dataframe/column.cpp [hunk 17]
```diff
@@ -34,24 +34,10 @@ gdf_error gdf_mask_concat(gdf_valid_type *output_mask,
...
- *             validity bitmasks
- * @param[in] columns_to_concat[] The columns to concatenate
- * @param[in] num_columns The number of columns to concatenate
- * 
- * @return gdf_error GDF_SUCCESS upon completion; GDF_DATASET_EMPTY if any data
- *         pointer is NULL, GDF_COLUMN_SIZE_MISMATCH if the output column size
- *         != the total size of the input columns; GDF_DTYPE_MISMATCH if the
- *         input columns have different datatypes.
- * ---------------------------------------------------------------------------**/
+ */
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
// Concatenates multiple gdf_columns into a single, contiguous column,
// including the validity bitmasks.
```

File: cpp/src/dataframe/column.cpp [hunk 18]
```diff
@@ -176,18 +148,11 @@ gdf_error gdf_column_view(gdf_column *column,
...
  *
- * @param[out] column The output column.
- * @param[in] data Pointer to data.
- * @param[in] valid Pointer to validity bitmask for the data.
- * @param[in] size Number of rows in the column.
- * @param[in] dtype Data type of the column.
- * @param[in] null_count The number of non-valid elements in the validity bitmask
- * @return gdf_error Returns GDF_SUCCESS upon successful creation.
- * ---------------------------------------------------------------------------**/
+ */
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
// Create a GDF column given data and validity bitmask pointers, size, and
// datatype, and count of null (non-valid) elements
```

File: cpp/src/dataframe/column.cpp [hunk 19]
```diff
@@ -205,12 +170,7 @@ gdf_error gdf_column_view_augmented(gdf_column *column,
 	return GDF_SUCCESS;
 }
 
-/** ---------------------------------------------------------------------------*
- * @brief Free the CUDA device memory of a gdf_column
- *
- * @param[inout] column Data and validity bitmask pointers of this column will be freed
- * @return gdf_error GDF_SUCCESS or GDF_ERROR if there is an error freeing the data
- * ---------------------------------------------------------------------------**/
+// @brief Free the CUDA device memory of a gdf_column
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
// Free the CUDA device memory of a gdf_column
```

File: cpp/src/dataframe/column.cpp [hunk 20]
```diff
@@ -228,14 +188,8 @@ namespace{
...
-/** ---------------------------------------------------------------------------*
- * @brief Get the byte width of a column
- *
- * @param[in] col The input column
- * @param[out] width The data type size of col
- * @return gdf_error GDF_SUCCESS, or GDF_UNSUPPORTED_DTYPE if col has an invalid
- *         datatype
- * ---------------------------------------------------------------------------**/
+
+//@brief  returns the byte width of the data type of the gdf_column
```
Comment (harrism) [inline cpp/src/dataframe/column.cpp]:
```suggestion
//  returns the byte width of the data type of the gdf_column
```

**Comment by harrism** [comment]
rerun tests

File: cpp/include/cudf/functions.h [hunk 21]
```diff
@@ -89,7 +89,8 @@ gdf_error gdf_column_view(gdf_column *column, void *data, gdf_valid_type *valid,
  * @param[in] valid Pointer to validity bitmask for the data.
  * @param[in] size Number of rows in the column.
  * @param[in] dtype Data type of the column.
- * @param[in] null_count The number of non-valid elements in the validity bitmask
+ * @param[in] null_count The number of non-valid elements in the validity bitmask.
+ * @param[in] extra_info see gdf_dtype_extra_info. Extra data for column description.
```
Comment (harrism) [inline cpp/include/cudf/functions.h]: For future reference, use @see to get doxygen to create an HTML link to the thing you reference.  Won't let it block this PR though.

**Pull request merged**